lib/babel-assertion-visitor.js

```diff
@@ -6,40 +6,34 @@ var babelgen = require('babel-generator');
 var define = require('define-properties');
 var toBeCaptured = require('./to-be-captured');
 var toBeSkipped = require('./to-be-skipped');
-var helperCode = [
-    '(function () {',
-    '    var captured = [];',
-    '    function _capt (value, espath) {',
-    '        captured.push({value: value, espath: espath});',
-    '        return value;',
-    '    }',
-    '    function _expr (value, args) {',
-    '        var source = {',
-    '            content: args.content,',
-    '            filepath: args.filepath,',
-    '            line: args.line',
-    '        };',
-    '        if (args.generator) {',
-    '            source.generator = true;',
-    '        }',
-    '        if (args.async) {',
-    '            source.async = true;',
-    '        }',
-    '        return {',
-    '            powerAssertContext: {',
-    '                value: value,',
-    '                events: captured',
-    '            },',
-    '            source: source',
-    '        };',
-    '    }',
-    '    return {',
-    '        _capt: _capt,',
-    '        _expr: _expr',
-    '    };',
-    '});'
-].join('\n');
-
+var helperCode = '(' + (function () {
+      function PowerAssertRecorder() {
+          this.captured = [];
+      }
+
+      PowerAssertRecorder.prototype._capt = function _capt (value, espath) {
+          this.captured.push({value: value, espath: espath});
+          return value;
+      };
+
+      PowerAssertRecorder.prototype._expr = function _expr (value, args) {
+          return {
+              powerAssertContext: {
+                  value: value,
+                  events: this.captured
+              },
+              source: {
```

Comment:
Why can't this just be `source: args`? I did it this way to mimic what you had before, but I am not entirely sure why it's necessary to create a safe copy of `args` here.